modules/ingest-geoip/src/internalClusterTest/java/org/elasticsearch/ingest/geoip/GeoIpDownloaderIT.java

```diff
@@ -283,6 +285,42 @@ public void testGeoIpDatabasesDownloadNoGeoipProcessors() throws Exception {
         }, 2, TimeUnit.MINUTES);
     }
 
+    public void testDoNotDownloadDatabaseOnPipelineCreation() throws Exception {
+        assumeTrue("only test with fixture to have stable results", getEndpoint() != null);
+        String pipelineId = randomIdentifier();
+
+        // Removing databases from tmp dir. So we can test the downloader.
+        deleteDatabasesInConfigDirectory();
+
+        // Enabling the downloader.
+        updateClusterSettings(Settings.builder().put(GeoIpDownloaderTaskExecutor.ENABLED_SETTING.getKey(), true));
+
+        // Creating a pipeline containing a geo processor with lazy download enable.
+        // Download should not be triggered and task state should stay null.
+        putGeoIpPipeline(pipelineId, true);
```

Comment:
Shouldn't this be false? We want to _not_ download on pipeline creation for this test right?

Reply:
And related, we probably ought to have a test that sets the value to true and checks that it does download right? Like
```
        putGeoIpPipeline(pipelineId, true);
        assertBusy(() -> assertNotNull(getTask().getState()));
```